metrics/local.go

```diff
@@ -265,15 +265,18 @@ func (l *localGauge) Update(value int64) {
 
 // LocalFactory stats factory that creates metrics that are stored locally
 type LocalFactory struct {
-	localBackend *LocalBackend
+	LocalBackend *LocalBackend
 	namespace    string
 	tags         map[string]string
 }
 
 // NewLocalFactory returns a new LocalMetricsFactory
-func NewLocalFactory(lb *LocalBackend) Factory {
+func NewLocalFactory(lb *LocalBackend) *LocalFactory {
```

Comment:
I would take the reporting frequency as argument and make the backend purely an implementation detail